### PR TITLE
create admin checkin page

### PR DESCRIPTION
**Pull request by menthy-wu**
image
try these qr codes
image
image

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -27,6 +27,12 @@ const nextConfig = {
         port: "",
         pathname: "/a/**",
       },
+      {
```
Comment (shahdivyank) [inline next.config.js]: this is the same as the config above, you should be able to combine them

File: package.json [hunk 2]
```diff
@@ -22,15 +22,18 @@
     "eslint-config-prettier": "^8.8.0",
     "firebase": "^10.0.0",
     "firebase-admin": "^11.10.1",
+    "i": "^0.3.7",
```
Comment (shahdivyank) [inline package.json]: what's this

File: package.json [hunk 3]
```diff
@@ -22,15 +22,18 @@
...
     "next": "^13.4.9",
     "next-auth": "^4.22.1",
+    "npm": "^9.8.1",
```
Comment (shahdivyank) [inline package.json]: what's this

File: src/components/Admin/DropDown.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    );
+  };
+  return (
+    <Dropdown className="w-full m-0 bg-white rounded-3xl">
+      <Dropdown.Toggle as={Toggle}>{option}</Dropdown.Toggle>
+      <Dropdown.Menu
+        as={Menu}
+        className="w-full bg-white border-none !rounded-b-xl !rounded-t-none border-2 border-black p-0 !z-10 !-mt-4 pt-4"
+      >
+        {/* <Dropdown.Menu className="w-full bg-white border-none !rounded-b-xl !rounded-t-none border-2 border-black p-0 !z-10 !-mt-4 pt-4"> */}
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: remove commented out code

File: src/components/Admin/DropDown.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    return (
+      <div className={className}>
+        <input
+          autoFocus
+          className="mx-3 my-2 w-fit ring-0 outline-none"
+          placeholder="Type to filter..."
+          onChange={(e) => setValue(e.target.value)}
+          value={value}
+        />
+        {React.Children.toArray(children).filter(
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: what is this 💀

File: next.config.js (hunk 1 above)
Comment (menthy-wu) [reply next.config.js]: no i tried, and they are different, one is a, one is a-

File: package.json (hunk 2 above)
Comment (menthy-wu) [reply package.json]: oh wtf lol

File: src/components/Admin/DropDown.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      <button
+        onClick={onClick}
+        className="relative z-50 bg-white text-black w-full border-2 rounded-full border-black justify-between flex items-center px-3 py-1"
+      >
+        {children}
+        <RiArrowDownSLine />
+      </button>
+    );
+  };
+  const Menu = ({ children, className }) => {
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: put these outside the component

File: src/components/Admin/DropDown.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from "react";
+import Dropdown from "react-bootstrap/Dropdown";
+import { RiArrowDownSLine } from "react-icons/ri";
+
+const DropDown = ({ options, setOption, option }) => {
+  const [value, setValue] = useState("");
+  const Toggle = ({ children, onClick }) => {
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: put outside the component

File: src/components/Admin/DropDown.jsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        <input
+          autoFocus
+          className="mx-3 my-2 w-fit ring-0 outline-none"
+          placeholder="Type to filter..."
+          onChange={(e) => setValue(e.target.value)}
+          value={value}
+        />
+        {children.filter(
+          (child) =>
+            !value || child.props.children.toLowerCase().startsWith(value)
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: do we rlly need to do child.props.children, surely there is a better way

File: src/components/Admin/DropDown.jsx [hunk 9]
```diff
@@ -12,30 +12,36 @@ const Toggle = ({ children, onClick }) => {
...
       <input
         autoFocus
         className="mx-3 my-2 w-fit ring-0 outline-none"
         placeholder="Type to filter..."
-        onChange={(e) => setValue(e.target.value)}
+        onChange={(e) => {
+          setValue(e.target.value);
+          setOptions(
+            options.filter(
+              (option) => !value || option.toLowerCase().startsWith(value)
```
Comment (shahdivyank) [inline src/components/Admin/DropDown.jsx]: do we need the toLowerCase(), we should be able to prevent this situation since its done by us only

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**